core/src/main/java/org/sonarsource/sonarlint/core/container/storage/StoragePaths.java

```diff
@@ -118,7 +120,32 @@ public Path getQProfilesPath() {
   }
 
   public Path getActiveRulesPath(String qProfileKey) {
-    return getGlobalStorageRoot().resolve(ACTIVE_RULES_FOLDER).resolve(encodeForFs(qProfileKey) + ".pb");
+    String filename = tryConvertBase64UuidToString(qProfileKey);
+    return getGlobalStorageRoot().resolve(ACTIVE_RULES_FOLDER).resolve(encodeForFs(filename) + ".pb");
+  }
+
+  /**
+   * Base64 encoding used by recent SonarQube for quality profile keys are not playing well with case insensitive FS. There is a risk
+   * of having two quality profiles keys collisions.
+   * If the key is a valid base 64 encoded string, we will reencode it as hexa instead (a bit longer, but resistant to case insensitive FS). 
+   */
+  static String tryConvertBase64UuidToString(String qProfileKey) {
+    try {
+      byte[] decoded = Base64.getUrlDecoder().decode(qProfileKey);
```

Comment:
I wonder if we had to decode the key instead of simply transforming to hex?

Reply:
Pros of your approach: works consistently for all strings
Cons: string will be a bit longer
How to choose ?

Reply:
no idea :) let's leave as it is

Reply:
Finally I simplified everything and use hex encoding for all strings we are using as filename, not only quality profile keys.